xonsh/completers/tools.py

```diff
@@ -0,0 +1,33 @@
+import builtins
+
+
+def get_filter_function():
```

Comment:
This name doesn't make a lot of sense to me.  Also, wouldn't it be faster to have the two other functions defined globally and then simply return one or the other.  These get redefined each time this function is called.